src/sdam/monitor.rs

```diff
@@ -184,7 +200,46 @@ impl HeartbeatMonitor {
         let result = match self.connection {
             Some(ref mut conn) => {
                 let command = is_master_command(self.client_options.server_api.as_ref());
-                run_is_master(command, conn).await
+                emit_event!(
+                    self,
+                    ServerHeartbeatStartedEvent {
+                        server_address: self.address.clone(),
+                    },
+                    handle_server_heartbeat_started_event
+                );
+
+                let start_time = Instant::now();
+                let response_result = conn.send_command(command, None).await;
+                let end_time = Instant::now();
+
+                let round_trip_time = end_time.duration_since(start_time);
+
+                match response_result {
```

Comment:
(cc @patrickfreed) Unfortunately matching on a tuple of `(response_result, response_result.and_then(|raw| raw.as_ref().as_is_master_response(round_trip_time)))` doesn't work here because both of the results in the tuple need ownership of the original error in `response_result`, which requires a clone.